tests/integrations/django/test_cache_module.py

```diff
@@ -0,0 +1,562 @@
+import pytest
+import os
+import random
+
+from django import VERSION as DJANGO_VERSION
+
+from werkzeug.test import Client
+
+try:
+    from django.urls import reverse
+except ImportError:
+    from django.core.urlresolvers import reverse
+
+import sentry_sdk
+from sentry_sdk.integrations.django import DjangoIntegration
+from sentry_sdk.integrations.django.caching import _get_span_description
+from tests.integrations.django.myapp.wsgi import application
+from tests.integrations.django.utils import pytest_mark_django_db_decorator
+
+
+DJANGO_VERSION = DJANGO_VERSION[:2]
+
+
+@pytest.fixture
+def client():
+    return Client(application)
+
+
+@pytest.fixture
+def use_django_caching(settings):
+    settings.CACHES = {
+        "default": {
+            "BACKEND": "django.core.cache.backends.locmem.LocMemCache",
+            "LOCATION": "unique-snowflake-%s" % random.randint(1, 1000000),
+        }
+    }
+
+
+@pytest.fixture
+def use_django_caching_with_middlewares(settings):
+    settings.CACHES = {
+        "default": {
+            "BACKEND": "django.core.cache.backends.locmem.LocMemCache",
+            "LOCATION": "unique-snowflake-%s" % random.randint(1, 1000000),
+        }
+    }
+    if hasattr(settings, "MIDDLEWARE"):
+        middleware = settings.MIDDLEWARE
+    elif hasattr(settings, "MIDDLEWARE_CLASSES"):
+        middleware = settings.MIDDLEWARE_CLASSES
+    else:
+        middleware = None
+
+    if middleware is not None:
+        middleware.insert(0, "django.middleware.cache.UpdateCacheMiddleware")
+        middleware.append("django.middleware.cache.FetchFromCacheMiddleware")
+
+
+@pytest.fixture
+def use_django_caching_with_port(settings):
+    settings.CACHES = {
+        "default": {
+            "BACKEND": "django.core.cache.backends.dummy.DummyCache",
+            "LOCATION": "redis://username:password@127.0.0.1:6379",
+        }
+    }
+
+
+@pytest.fixture
+def use_django_caching_with_cluster(settings):
+    settings.CACHES = {
+        "default": {
+            "BACKEND": "django.core.cache.backends.dummy.DummyCache",
+            "LOCATION": [
+                "redis://127.0.0.1:6379",
+                "redis://127.0.0.2:6378",
+                "redis://127.0.0.3:6377",
+            ],
+        }
+    }
+
```

Comment:
Please also add a test with URL lacking a port, e.g. https://example.com, as requested in a previous comment

Reply:
done